backend/src/v2/compiler/argocompiler/argo.go

```diff
@@ -33,6 +33,7 @@ type Options struct {
 	LauncherImage string
 	// optional
 	DriverImage string
+	runnerImage string
```

Comment:
nit: make this uppercase to make the field public?
Also change cmd/compiler to expose the flag and update v2 sample test to build the new runner image.